### PR TITLE
Updated Chirpstack from V3 to V4

### PR DESCRIPTION
**Pull request by augusthjerrild**
ServiceProfile and networkserver is removed from V4.

### CODE REVIEW

File: src/app/profiles/device-profiles/device-profiles-list/device-profiles-list.component.ts [hunk 1]
```diff
@@ -25,6 +24,7 @@ export class DeviceProfilesListComponent implements OnInit, OnDestroy {
   public errorMessages: any;
   deleteDialogSubscription: Subscription;
   errorTitle: string;
+  isDeleting = true;
```
Comment (fcv-iteratorIt) [inline src/app/profiles/device-profiles/device-profiles-list/device-profiles-list.component.ts:27]: Don't default true

File: src/app/profiles/device-profiles/device-profiles-list/device-profiles-list.component.ts [hunk 2]
```diff
@@ -42,6 +42,7 @@ export class DeviceProfilesListComponent implements OnInit, OnDestroy {
         this.errorTitle = translations['PROFILES.DELETE-FAILED'];
       });
     this.getDeviceProfiles();
+    this.isDeleting = false;
```
Comment (fcv-iteratorIt) [inline src/app/profiles/device-profiles/device-profiles-list/device-profiles-list.component.ts:45]: No need to both default when declaring and setting in onInit

File: src/app/profiles/device-profiles/device-profiles-list/device-profiles-list.component.ts [hunk 3]
```diff
@@ -73,6 +74,7 @@ export class DeviceProfilesListComponent implements OnInit, OnDestroy {
       this.deleteDialogSubscription = this.deleteDialogService.showSimpleDialog().subscribe(
         (response) => {
           if (response) {
+            this.isDeleting = true;
```
Comment (fcv-iteratorIt) [inline src/app/profiles/device-profiles/device-profiles-list/device-profiles-list.component.ts:77]: Shouldn't is deleting be set to true before we start the api call?

File: src/app/shared/services/chirpstack-gateway.service.ts [hunk 4]
```diff
@@ -1,87 +1,94 @@
...
-      );
-  }
+    public get(id: string, params = {}): Observable<GatewayResponse> {
+        return this.restService.get(this.chripstackGatewayUrl, params, id).pipe(
+            map((response: GatewayResponse) => {
+                response.gateway.internalOrganizationName = this.sharedVariableService
+                    .getOrganizationInfo()
+                    .find((org) => org.id === response.gateway.internalOrganizationId)?.name;
+                // Move createdat and updatedat to next level ease the use.
+                response.gateway.tagsString = JSON.stringify(response.gateway.tags)
```
Comment (fcv-iteratorIt) [inline src/app/shared/services/chirpstack-gateway.service.ts:38]: Does created/updated at still display properly now that it is not moved here?

**Review by fcv-iteratorIt — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -87,6 +87,9 @@
     "xml2js": "^0.6.2",
     "babel-plugin-polyfill-corejs3": "0.8.6",
     "babel-plugin-polyfill-corejs2": "0.4.6",
-    "babel-plugin-polyfill-regenerator": "0.5.3"
+    "babel-plugin-polyfill-regenerator": "0.5.3",
+    "ngx-monaco-editor-v2": {
```
[the remark this answers is not in the document]
Comment (augusthjerrild) [reply package.json:91]: Weird, I earlier got some legacy-peer-deps error, therefore i added this. Now i dont. Reverted :-)

File: src/app/profiles/device-profiles/device-profiles-list/device-profiles-list.component.ts (hunk 1 above)
Comment (augusthjerrild) [reply src/app/profiles/device-profiles/device-profiles-list/device-profiles-list.component.ts:27]: I will remove this isDeleting stuff again. Took inspiration from other places in the code, but not sure it's needed anyways. And not sure it was actually working.

File: src/app/shared/services/chirpstack-gateway.service.ts (hunk 4 above)
Comment (augusthjerrild) [reply src/app/shared/services/chirpstack-gateway.service.ts:38]: Yes, it works :-)

File: src/app/gateway/gateway-table/gateway-table.component.ts [hunk 6]
```diff
@@ -113,7 +185,7 @@ export class GatewayTableComponent implements AfterViewInit, OnDestroy {
 
     lastActive(gateway: Gateway): string {
         if (gateway?.lastSeenAt) {
-            const date = gateway.lastSeenAt ?? convertToDateFromTimestamp(gateway.lastSeenAt);
+            const date = gateway.lastSeenAt ?? gateway.lastSeenAt
```
Comment (fcv-iteratorIt) [inline src/app/gateway/gateway-table/gateway-table.component.ts:188]: lol

**Review by fcv-iteratorIt — APPROVED** [review]

File: src/app/profiles/device-profiles/device-profiles-list/device-profiles-list.component.ts [hunk 7]
```diff
@@ -11,101 +11,91 @@ import { environment } from '@environments/environment';
...
-            this.deviceProfileService.delete(id).subscribe((response) => {
-              console.log(response);
-              if (response.ok) {
-                this.getDeviceProfiles();
-              } else {
-                if (response?.error?.message === 'Internal server error') {
-                  this.errorMessages = 'Internal server error';
-                  return;
+    deleteDeviceProfile(id: string) {
+        if (id) {
```
[the remark this answers is not in the document]
Comment (fcv-iteratorIt) [reply src/app/profiles/device-profiles/device-profiles-list/device-profiles-list.component.ts]: Nvm I couldn't stand it and fixed it now

**Pull request merged**